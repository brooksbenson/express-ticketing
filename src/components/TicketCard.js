import React from 'react';
import moment from 'moment';

const colorUrgency = urgency => {
  switch (urgency) {
    case 'high':
      return '#ff0000';
    case 'medium':
      return '#ff7400';
    case 'low':
      return '#ffc100';
  }
};

export default props => (
  <div
    className={props.className}
    onClick={() => {
      props.history.push(`ticket/${props.ticketKey}`);
    }}
  >
    <div className={`${props.className}__top`}>
      <h3>{props.accountName}</h3>
      <button
        className="urgency"
        style={{ background: colorUrgency(props.urgency) }}
      />
    </div>
    <p className="title">{props.title}</p>
    <p className="date">{moment(props.date).format('MM-DD-YY [at] h:ma')}</p>
  </div>
);
