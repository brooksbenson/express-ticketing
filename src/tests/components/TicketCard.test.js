import React from 'react';
import { shallow } from 'enzyme';
import TicketCard from '../../components/TicketCard';

const red = '#ff0000';
const orange = '#ff7400';
const yellow = '#ffc100';

const ticket = {
  ticketKey: 'jfkdajf;',
  accountName: 'Walmart',
  title: 'Shipping delay',
  date: 1231234442324,
  urgency: 'high'
};

const className = 'ticket-card';

let wrapper;
let history;

beforeEach(() => {
  history = {
    push: jest.fn()
  };
  wrapper = shallow(
    <TicketCard className={className} history={history} {...ticket} />
  );
});

test('TicketCard should match snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});

test('TicketCard should assign the wrapper className to the className passed as a prop', () => {
  expect(wrapper.hasClass(className)).toBeTruthy();
});

test('TicketCard should pass red hex color to .urgency when urgency is high', () => {
  const style = wrapper.find('.urgency').prop('style');
  expect(style.background).toBe(red);
});

test('TicketCard should pass orange hex color to .urgency when urgency is medium', () => {
  wrapper.setProps({ urgency: 'medium' });
  const style = wrapper.find('.urgency').prop('style');
  expect(style.background).toBe(orange);
});

test('TicketCard should pass yellow hex color to .urgency when urgency is low', () => {
  wrapper.setProps({ urgency: 'low' });
  const style = wrapper.find('.urgency').prop('style');
  expect(style.background).toBe(yellow);
});

test('TicketCard should invoke the history.push method and pass the ticket key as a parameter when wrapper is clicked on', () => {
  wrapper.simulate('click');
  expect(history.push).toHaveBeenCalledWith(`ticket/${ticket.ticketKey}`);
});
