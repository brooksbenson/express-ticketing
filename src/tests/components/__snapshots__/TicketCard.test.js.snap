// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketCard should match snapshot 1`] = `
<div
  className="ticket-card"
  onClick={[Function]}
>
  <div
    className="ticket-card__top"
  >
    <h3>
      Walmart
    </h3>
    <div
      className="urgency"
      style={
        Object {
          "background": "#ff0000",
        }
      }
    />
  </div>
  <p>
    Shipping delay
  </p>
  <p>
    1231234442324
  </p>
</div>
`;
